native/packages/react-native-bpk-component-text/src/BpkText.js

```diff
@@ -34,72 +30,57 @@ const tokens = Platform.select({
 
 const TEXT_STYLES = ['xs', 'sm', 'base', 'lg', 'xl', 'xxl'];
 
-const styleForTextStyle = (textStyle) => {
+const getAndroidFontFamily = fontWeight =>
+  (fontWeight === '500' ? 'sans-serif-medium' : 'sans-serif');
+
+const getStyleByTextStyle = (textStyle) => {
   const camelCasedStyle = textStyle[0].toUpperCase() + textStyle.slice(1);
 
-  const determineAndroidFontFamily = (fontWeight) => {
-    switch (fontWeight) {
-      case '500':
-        return 'sans-serif-medium';
-      case '400':
-      default:
-        return 'sans-serif';
-    }
-  };
-
-  return {
-    fontSize: tokens[`text${camelCasedStyle}FontSize`],
-    lineHeight: tokens[`text${camelCasedStyle}LineHeight`],
-    fontWeight: Platform.OS === 'ios' ? tokens[`text${camelCasedStyle}FontWeight`] : undefined,
-    color: tokens.colorGray900,
-    fontFamily: Platform.OS === 'android' ?
-      determineAndroidFontFamily(tokens[`text${camelCasedStyle}FontWeight`]) : 'System',
-  };
+  const color = tokens.colorGray700;
+  const fontSize = tokens[`text${camelCasedStyle}FontSize`];
+  const lineHeight = tokens[`text${camelCasedStyle}LineHeight`];
+  const fontWeight = Platform.OS === 'ios' ? tokens[`text${camelCasedStyle}FontWeight`] : undefined;
+  const androidFontFamily = getAndroidFontFamily(tokens[`text${camelCasedStyle}FontWeight`]);
+  const fontFamily = Platform.OS === 'android' ? androidFontFamily : 'System';
+
+  return { fontSize, lineHeight, fontWeight, color, fontFamily };
 };
 
 const styles = StyleSheet.create({
-  xs: {
-    ...styleForTextStyle('xs'),
-  },
-  sm: {
-    ...styleForTextStyle('sm'),
-  },
-  base: {
-    ...styleForTextStyle('base'),
-  },
-  lg: {
-    ...styleForTextStyle('lg'),
-  },
-  xl: {
-    ...styleForTextStyle('xl'),
-  },
-  xxl: {
-    ...styleForTextStyle('xxl'),
-  },
+  xs: { ...getStyleByTextStyle('xs') },
+  sm: { ...getStyleByTextStyle('sm') },
+  base: { ...getStyleByTextStyle('base') },
+  lg: { ...getStyleByTextStyle('lg') },
+  xl: { ...getStyleByTextStyle('xl') },
+  xxl: { ...getStyleByTextStyle('xxl') },
 });
 
 const BpkText = (props) => {
-  const { children, textStyle, style: innerStyle, ...rest } = props;
-  const style = styles[textStyle];
-  const finalStyles = [style];
+  const { children, textStyle, style, bold, ...rest } = props;
 
-  if (innerStyle) {
-    finalStyles.push(innerStyle);
+  const fontWeightStyle = bold ? { fontWeight: '600' } : {};
```

Comment:
In my buttons branch I've [added a token](https://github.com/Skyscanner/backpack/pull/83/files#diff-07fc94f9b6facc08135d030f1193ca7bR216) that corresponds to `fontWeight: 600`, so once that's merged, this could change to use that.